core/core.go

```diff
@@ -94,21 +94,29 @@ type WalletConfig interface {
 	IsUsable() error
 }
 
-// A SignerSignatures object stores signer signatures indexed by signer and subdigest.
-type SignerSignatures map[common.Address]map[common.Hash]SignerSignature
+// A SignerSignatures object stores SignerSignature and SignerEIP6492Signature objects indexed by signer and subdigest.
+type SignerSignatures map[common.Address]map[common.Hash]interface{}
```

Comment:
Hmm I am not sure if this is better, at the end of the day ERC6492 signatures can always be detected by looking at the prefix, which can be done after obtaining these signatures.

Sorry, I think I am going to ask you to dismiss my previous comment and leave it as it was before.

Reply:
Yeah I agree. Would've made sense for any other language but Go inheritance is yuck...